homeassistant/components/apple_tv/__init__.py

```diff
@@ -337,13 +329,36 @@ async def _connect(self, conf):
         self._dispatch_send(SIGNAL_CONNECTED, self.atv)
         self._address_updated(str(conf.address))
 
+        await self._async_setup_device_registry()
+
         self._connection_attempts = 0
         if self._connection_was_lost:
             _LOGGER.info(
                 'Connection was re-established to Apple TV "%s"', self.atv.service.name
             )
             self._connection_was_lost = False
 
+    async def _async_setup_device_registry(self):
+        attrs = {
+            "identifiers": {(DOMAIN, self.config_entry.unique_id)},
+            "manufacturer": "Apple",
+            "name": self.config_entry.data.get(CONF_NAME),
+        }
+
+        if self.atv:
+            dev_info = self.atv.device_info
+
+            attrs["model"] = "Apple TV " + dev_info.model.name.replace("Gen", "")
+            attrs["sw_version"] = dev_info.version
+
+            if dev_info.mac:
+                attrs["connections"] = {(dr.CONNECTION_NETWORK_MAC, dev_info.mac)}
+
+        device_registry = await dr.async_get_registry(self.hass)
+        device_registry.async_get_or_create(
```

Comment:
Can we report this just for the media player entity instead, by implementing `device_info` entity property?

Reply:
Since all entities are added before a connection to the device is made, the device info has already been pulled from the entities even before it has been fetched from the device. So I don't think it's possible?

Reply:
Ok.